packages/widget/src/controllers/transfers/fungibleTokenTransfer.ts

```diff
@@ -0,0 +1,284 @@
+    if (this.sourceNetwork && !this.selectedResource) {
+      //filter resources
+      void this.filterDestinationNetworksAndResources(this.sourceNetwork);
+      return;
+    }
+    void this.buildTransactions();
+    this.host.requestUpdate();
+  };
+
+  onResourceSelected = (resource: Resource, amount: BigNumber): void => {
+    this.selectedResource = resource;
+    this.resourceAmount = amount;
+    void this.buildTransactions();
+    this.host.requestUpdate();
+  };
+
+  onDestinationAddressChange = (address: string): void => {
+    this.destinatonAddress = address;
+    if (this.destinatonAddress.length === 0) {
+      this.pendingEvmApprovalTransactions = [];
+      this.pendingEvmTransferTransaction = undefined;
+    }
+    void this.buildTransactions();
+    this.host.requestUpdate();
+  };
+
+  getTransferState(): FungibleTransferState {
+    if (!this.sourceNetwork) {
+      return FungibleTransferState.MISSING_SOURCE_NETWORK;
+    }
+    if (!this.destinationNetwork) {
+      return FungibleTransferState.MISSING_DESTINATION_NETWORK;
+    }
+    if (!this.selectedResource) {
+      return FungibleTransferState.MISSING_RESOURCE;
+    }
+    if (this.resourceAmount.eq(0)) {
+      return FungibleTransferState.MISSING_RESOURCE_AMOUNT;
+    }
+    if (this.destinatonAddress === '') {
+      return FungibleTransferState.MISSING_DESTINATION_ADDRESS;
+    }
+    if (
+      !this.walletContext.value?.evmWallet &&
+      !this.walletContext.value?.substrateWallet
+    ) {
+      return FungibleTransferState.WALLET_NOT_CONNECTED;
+    }
+    if (
+      this.sourceNetwork.type === Network.EVM &&
+      this.walletContext.value?.evmWallet?.providerChainId !==
+        this.sourceNetwork.chainId
+    ) {
+      return FungibleTransferState.WRONG_CHAIN;
+    }
+    if (this.waitingUserConfirmation) {
+      return FungibleTransferState.WAITING_USER_CONFIRMATION;
+    }
+    if (this.waitingTxExecution) {
+      return FungibleTransferState.WAITING_TX_EXECUTION;
+    }
+    if (this.transferTransactionId) {
+      return FungibleTransferState.COMPLETED;
+    }
+    if (this.pendingEvmApprovalTransactions.length > 0) {
+      return FungibleTransferState.PENDING_APPROVALS;
+    }
+    if (this.pendingEvmTransferTransaction) {
+      return FungibleTransferState.PENDING_TRANSFER;
+    }
+    return FungibleTransferState.UNKNOWN;
+  }
+
+  executeTransaction(): void {
+    if (!this.sourceNetwork) {
+      return;
+    }
+    switch (this.sourceNetwork.type) {
+      case Network.EVM:
+        {
+          void this.executeNextEvmTransaction();
+        }
+        break;
+      case Network.SUBSTRATE:
+        {
+          //TODO: add substrate logic
+        }
+        break;
+      default:
+        throw new Error('Unsupported network type');
+    }
+  }
+
+  getExplorerLink(): string {
+    if (this.env === Environment.MAINNET) {
+      return `https://scan.buildwithsygma.com/transfer/${this.transferTransactionId}`;
```

Comment:
nIt: although scan urls are not going to change, so having them here is ok, perhaps we could move those to a constant file